src/test/java/com/quorum/gauge/PENetworkUpgrade.java

```diff
@@ -163,6 +163,23 @@ public void nodeStateIs(String component, String node, String state) {
         assertThat(containerState).isEqualTo(state);
     }
 
+    @Step("Wait for <component> state in <node> to be <state>")
+    public void waitForNodeState(String component, String node, String state) {
+        String containerId = getComponentContainerId(component, node);
+        String containerState = infraService.wait(containerId).blockingFirst() ? "up" : "down";
+        int count = 20;
+        while (!containerState.equals(state) && count > 0){
+            try {
+                Thread.sleep(3000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
Please throw `RuntimeException` to politely exit the execution

Reply:
I've wrapped the InterruptedException in a RuntimeException (and added some extra details).